tools/zsh/_e

```diff
@@ -0,0 +1,101 @@
+#compdef e
+
+# For oh-my-zsh, install this in $ZSH/completions/ or ~/.oh-my-zsh/completions/
+# (You may need to mkdir that completions/ directory.)
+#
+# If you're not using oh-my-zsh, you may need to add this to your ~/.zshrc:
+# source /path/to/_e
+# autoload -U compinit && compinit
+#
+# To force zsh to regenerate completions, you can `rm -f ~/.zcompdump`
+
+declare -a __e_options=()
+function __e_get_cmd_options() {
+  local -a options=()
+  "$@" --help | \
+    awk '/^Options:/{flag=1;next}/^$/{flag=0}flag' | \
+    sed 's/  /;/g' | cut -f2 -d';' | \
+    sed 's/^-[a-zA-Z0-9], //g' | \
+    cut -f1 -d' ' | \
+    while read -r option; do
+      options+=($option)
+    done
+  __e_options=($options[@])
+}
+
+declare -a __e_subcommands=()
+function __e_get_cmd_subcommands() {
+  local -a subcommands=()
+  "$@" --help | \
+  awk '/^Commands:/{flag=1;next}/^$/{flag=0}flag' | \
+  sed 's/  /;/g' | cut -f2 -d';' | \
+  sed 's/|/\n/' | \
+  cut -f1 -d' ' | \
+  while read -r subcommand; do
+    subcommands+=($subcommand)
+  done
+  __e_subcommands=("$subcommands[@]")
+}
```

Comment:
i'd really like some comments for the code in this file, it's somewhat opaque

Reply:
These autocomplete scripts are _all_ flaming nightmares. I'll add some comments but it's still going to be somewhat opaque :smiley: 

Added 691fdeb4803b682a2a224742ed767c1d5877371d